aptos-move/aptos-replay-benchmark/src/generator.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{block::Block, overrides::OverrideConfig, workload::Workload};
+use aptos_move_debugger::aptos_debugger::AptosDebugger;
+use aptos_types::transaction::{Transaction, Version};
+use std::{
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    time::Instant,
+};
+
+pub struct BenchmarkGenerator {
```

Comment:
nit: I wouldn't define this struct - seems like we can have free function for generation and otherwise use Arc<BenchmarkGeneratorContext>, less hoops to go through and the abstraction as used now doesn't add much (and hides the Arc)

Reply:
Removed it, changed to `BenchmarkGenerator::generate_blocks(...)` instead.